### PR TITLE
Simple Eclipses

### PR DESCRIPTION
**Pull request by Schneegans**
New Features:
* Eclipse shadows cast by spherical bodies without incorporation of atmospheric effects
* Oren-Nayar shading for simple bodies
* Higher-resolution Jupiter texture
* HDR image file format support for the texture loader
* Gamma-only tone mapping operator

The eclipse shadow texture generator will be added in another pull request.

### CODE REVIEW

**Comment by JonasGilg** [comment]
There seems to be an error towards the start and end of eclipses. From one second to another the shadow disappears completely.
Below are two example pictures one second after the other:
Eclipse_2022-05-16T06-22-36
Eclipse_2022-05-16T06-22-37

If you toggle between them in an image viewer or in CosmoScout you can see, that the shadow diappears abruptly, instead of smoothly.

**Comment by JonasGilg** [comment]
> If you toggle between them in an image viewer or in CosmoScout you can see, that the shadow diappears abruptly, instead of smoothly.

Another example would be 2006-03-29 12:17:51 + 12:17:52 over central Asia.

**Comment by Schneegans** [comment]
Thanks for discovering this! There was indeed a bug in the CPU-side identification of potential eclipse shadow casters (the Sun had a radius of zero..). Now it should be working!

File: src/cs-core/EclipseShadowReceiver.hpp [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+//                        Copyright: (c) 2019 German Aerospace Center (DLR)                       //
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef CS_CORE_ECLIPSE_SHADOW_RECEIVER_HPP
+#define CS_CORE_ECLIPSE_SHADOW_RECEIVER_HPP
+
+#include "cs_core_export.hpp"
+
+#include <glm/glm.hpp>
+#include <memory>
```
Comment (JonasGilg) [inline src/cs-core/EclipseShadowReceiver.hpp:13]:
```suggestion
#include <glm/glm.hpp>
#include <array>
#include <memory>
```

File: src/cs-core/Settings.hpp [hunk 2]
```diff
@@ -497,6 +503,12 @@ class CS_CORE_EXPORT Settings {
     /// This makes illumination calculations assume a fixed sun position in the current SPICE frame.
     /// Using the default value glm::dvec3(0.0) disables this feature.
     utils::DefaultProperty<glm::dvec3> pFixedSunDirection{glm::dvec3(0.0, 0.0, 0.0)};
+
+    /// This maps anchor names to eclipse textures.
+    std::optional<std::unordered_map<std::string, EclipseShadowMap>> mEclipseShadowMaps;
+
+    /// The eclipse shadow randering mode.
```
Comment (JonasGilg) [inline src/cs-core/Settings.hpp:510]:
```suggestion
    /// The eclipse shadow rendering mode.
```

**Review by JonasGilg — CHANGES_REQUESTED** [review]

File: plugins/csp-atmospheres/src/Plugin.cpp [hunk 3]
```diff
@@ -273,7 +273,8 @@ void Plugin::onLoad() {
       continue;
     }
 
-    auto atmosphere = std::make_shared<Atmosphere>(mPluginSettings, mAllSettings, settings.first);
+    auto atmosphere =
```
[the remark this answers is not in the document]
Comment (Schneegans) [reply plugins/csp-atmospheres/src/Plugin.cpp]: Fixed.

**Review by JonasGilg — APPROVED** [review]
Great work! I am really happy that this has reached the core of CosmoScout VR 🥳

**Pull request merged**